File: src/ca/qc/bdeb/inf203/model/powerups/PlanteUnlock.java
package ca.qc.bdeb.inf203.model.powerups;

import ca.qc.bdeb.inf203.model.Item;
import ca.qc.bdeb.inf203.model.Joueur;
import ca.qc.bdeb.inf203.model.PowerUp;
import java.awt.Point;
import java.awt.Rectangle;

/**
 * Power-up débloquant un type de plante.
 * @author Guillaume Riou, Nicolas Hurtubise
 */
public class PlanteUnlock extends PowerUp {

    private Item item;
    private int largeur = 65;
    private int hauteur = 62;
   
    public PlanteUnlock(Item item, Point position) {
        super(item.getNom());
        this.item = item;
        this.hitbox = new Rectangle(position.x, position.y, largeur, hauteur);
        this.animationFrameRate = 1;
        this.nbrImagesAnimation = 1;
    }

    @Override
    public void action() {
        Joueur.instance().debloquerItem(item);
    }
}

File: assets/js/model/powerups/plante-unlock.js
/**
 * Power-up débloquant un type de plante.
 * @author Guillaume Riou, Nicolas Hurtubise
 */
function PlanteUnlock(item, position) {
    PowerUp.apply(this);

    this.item = item;
    this.largeur = 65;
    this.hauteur = 62;

    this.hitbox = new Rectangle(position.x, position.y, this.largeur, this.hauteur);
    this.animationFrameRate = 1;
    this.nbrImagesAnimation = 1;

    this.sprites[navigator.Etat.ATTENTE] = new RepresentationImage(["powerups", item.nom]);
}

PlanteUnlock.prototype = new PowerUp();

PlanteUnlock.prototype.action = function() {
    navigator.Joueur.debloquerItem(this.item);
};
